.github/workflows/pulsar-ci.yaml

```diff
@@ -610,30 +602,135 @@ jobs:
+        with:
+          limit-access-to-actor: true
+
+      - name: Cache local Maven repository
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.m2/repository/*/*/*
+            !~/.m2/repository/org/apache/pulsar
+          key: ${{ runner.os }}-m2-dependencies-all-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-m2-dependencies-core-modules-${{ hashFiles('**/pom.xml') }}
+            ${{ runner.os }}-m2-dependencies-core-modules-
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'temurin'
+          java-version: 17
+
+      - name: Install gh-actions-artifact-client.js
+        uses: apache/pulsar-test-infra/gh-actions-artifact-client/dist@master
+
+      - name: Restore maven build results from Github artifact cache
+        run: |
+          cd $HOME
+          $GITHUB_WORKSPACE/build/pulsar_ci_tool.sh restore_tar_from_github_actions_artifacts pulsar-maven-repository-binaries
+
+      - name: Load docker image apachepulsar/pulsar-test-latest-version:latest from Github artifact cache
+        run: |
+          $GITHUB_WORKSPACE/build/pulsar_ci_tool.sh docker_load_image_from_github_actions_artifacts pulsar-test-latest-version-image
+
+      - name: Run setup commands
+        if: ${{ matrix.setup }}
+        run: |
+          ${{ matrix.setup }}
+
+      - name: Run system test group '${{ matrix.group }}'
+        run: |
+          ./build/run_integration_group.sh ${{ matrix.group }}
+
+      - name: print JVM thread dumps when cancelled
+        if: cancelled()
+        run: $GITHUB_WORKSPACE/build/pulsar_ci_tool.sh print_thread_dumps
+
+      - name: Aggregates all test reports to ./test-reports and ./surefire-reports directories
+        if: ${{ always() }}
+        uses: ./.github/actions/copy-test-reports
+
+      - name: Publish Test Report
+        uses: apache/pulsar-test-infra/action-junit-report@master
+        if: ${{ always() }}
+        with:
+          report_paths: 'test-reports/TEST-*.xml'
+          annotate_only: 'true'
+
+      - name: Upload container logs
+        uses: actions/upload-artifact@v3
+        if: ${{ !success() }}
+        continue-on-error: true
+        with:
+          name: System-${{ matrix.group }}-container-logs
+          path: tests/integration/target/container-logs
+          retention-days: 7
+
+      - name: Upload Surefire reports
+        uses: actions/upload-artifact@v3
+        if: ${{ !success() }}
+        with:
+          name: System-${{ matrix.name }}-surefire-reports
+          path: surefire-reports
+          retention-days: 7
+
+      - name: Wait for ssh connection when build fails
+        # ssh access is enabled for builds in own forks
+        uses: ./.github/actions/ssh-access
+        if: ${{ failure() && github.repository != 'apache/pulsar' }}
+        continue-on-error: true
+        with:
+          action: wait
+
+  flaky-system-tests:
```

Comment:
what is the exact reason for adding this? I would assume that it's possible to add matrix variables to cover this.

Reply:
yes but then the new step `pulsar-ci-checks-completed` needs these tests passing. Currently in the `.asf.yaml` they are not listed. So now it depends only on `system-tests` job and the result of this one doesn't impact the mergeablity of the pull

Reply:
oh yes you are right.